### PR TITLE
refactor(grafana): refactor grafana pvc

### LINKED ISSUE #371: When uninstalling SN Platform, Grafana PVC remains
If I do `helm install` of an SN-platform instance use that instance for awhile, and then `helm uninstall` it, everything is cleaned up except for one PersistentVolumeClaim and its associated volumes: the one used for Grafana.

Given an installation name of `pulsar-blt`, after uninstallation completes I still have a PVC named `pulsar-blt-sn-platform-grafana-data-pulsar-blt-sn-platform-grafana-0`.

### PR DESCRIPTION
**Pull request by ericsyh**
Signed-off-by: ericsyh <ericshenyuhao@outlook.com>

*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #371 

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

Refactor the Grafana PVC and fix the issue 371

### Modifications

* Added the pvc template for Grafana.
* Updated the Grafana statefulset template.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform/templates/grafana/grafana-pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# under the License.
+#
+
+{{- if .Values.monitoring.grafana }}
+{{- if and .Values.volumes.persistence .Values.grafana.volumes.persistence }}
+{{- if not (or .Values.grafana.volumes.data.storageClass .Values.grafana.volumes.data.storageClassName) }}
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: "{{ template "pulsar.fullname" . }}-{{ .Values.grafana.component }}-{{ .Values.grafana.volumes.data.name }}"
```
Comment (MarvinCai) [inline charts/sn-platform/templates/grafana/grafana-pvc.yaml:26]: use  `"{{ template "pulsar.grafana.volumes.pvc.name" . }}"` for simplicity?

Comment (ericsyh) [reply charts/sn-platform/templates/grafana/grafana-pvc.yaml:26]: yes, we can also fix in https://github.com/streamnative/charts/blob/master/charts/sn-platform/templates/prometheus/prometheus-pvc.yaml#L26

Comment (ericsyh) [reply charts/sn-platform/templates/grafana/grafana-pvc.yaml]: updated in https://github.com/streamnative/charts/pull/857/commits/58ad5f5e4511420056e1835ed28426a3a24ea135

**Comment by tuteng** [comment]
@MarvinCai PTAL

**Comment by ciiiii** [comment]
CI job failed because it was created from forks, sensitive data like secrets can't be used in such a situation.

Can you try to recreate this PR by a topic branch in this repo? @ericsyh 

[Logs](https://github.com/streamnative/charts/runs/8262757738?check_suite_focus=true)
```
By default, secrets are not passed to workflows triggered from forks, including Dependabot.
```

**Pull request closed without merging**